File: trunk/src/scala/org/codehaus/testdox/intellij/TestInterface.scala
package org.codehaus.testdox.intellij

import com.intellij.psi.PsiClass

class TestInterface(className: String, psiClass: PsiClass, editorApi: EditorApi, resolver: NameResolver)
    extends TestClass(className, psiClass, editorApi, resolver) {

  override def displayString = "<i>" + super.displayString + "</i>"

  override val icon = {
    if (psiElement != null)
      Icons.getIcon(Icons.INTERFACE_ICON)
    else
      Icons.getLockedIcon(Icons.INTERFACE_ICON)
  }
}

File: branches/pre-scala-migration/src/java/org/codehaus/testdox/intellij/TestInterface.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.psi.PsiClass;

public class TestInterface extends TestClass {

    public TestInterface(String className, PsiClass psiClass, EditorApi editorApi, NameResolver resolver) {
        super(className, psiClass, editorApi, resolver);
    }

    public String getDisplayString() {
        return new StringBuffer("<i>").append(super.getDisplayString()).append("</i>").toString();
    }

    public Icon getIcon() {
        return (getPsiElement() != null) ? IconHelper.getIcon(IconHelper.INTERFACE_ICON)
                                         : IconHelper.getLockedIcon(IconHelper.INTERFACE_ICON);
    }
}
